frontend/src/components/common/EditButton.tsx

```diff
@@ -20,12 +20,30 @@ export default function EditButton(props: EditButtonProps) {
   const { item, options = {} } = props;
   const [openDialog, setOpenDialog] = React.useState(false);
   const [visible, setVisible] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState<string>('');
   const location = useLocation();
 
-  const applyFunc = React.useCallback((newItem: KubeServiceAccount) => {
-    return item.update(newItem);
-  },
-  [item]);
+  function makeErrorMessage(err: any) {
+    const status: number = err.status;
+    switch (status) {
+      case 408:
+        return 'Conflicts when trying to perform operation (code 408).';
+      default:
+        return `Failed to perform operation: code ${status}`;
+    }
+  }
+
+  async function updateFunc(newItem: KubeServiceAccount) {
+    try {
+      await item.update(newItem);
+    } catch (err) {
+      setErrorMessage(makeErrorMessage(err));
+      setOpenDialog(true);
+      throw err;
```

Comment:
we are already in the catch block why do you want to throw err from here? 
it will break the app if i am correct

Reply:
We catch it here so we show the dialog, etc. then we throw it again so it gets caught and treated in the saga (so it displays a notification). This is usually called error propagation.

Reply:
cool if its caught in the saga than its fine